nessie/src/main/java/org/apache/iceberg/nessie/NessieViewOperations.java

```diff
@@ -0,0 +1,159 @@
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.nessie;
+
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicBoolean;
+import org.apache.iceberg.exceptions.AlreadyExistsException;
+import org.apache.iceberg.exceptions.NoSuchViewException;
+import org.apache.iceberg.io.FileIO;
+import org.apache.iceberg.relocated.com.google.common.collect.Maps;
+import org.apache.iceberg.view.BaseViewOperations;
+import org.apache.iceberg.view.ViewMetadata;
+import org.apache.iceberg.view.ViewMetadataParser;
+import org.projectnessie.client.http.HttpClientException;
+import org.projectnessie.error.NessieConflictException;
+import org.projectnessie.error.NessieNotFoundException;
+import org.projectnessie.model.Content;
+import org.projectnessie.model.ContentKey;
+import org.projectnessie.model.IcebergTable;
+import org.projectnessie.model.IcebergView;
+import org.projectnessie.model.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class NessieViewOperations extends BaseViewOperations {
+
+  private static final Logger LOG = LoggerFactory.getLogger(NessieViewOperations.class);
+
+  private final NessieIcebergClient client;
+  private final ContentKey key;
+  private final FileIO fileIO;
+  private final Map<String, String> catalogOptions;
+  private IcebergView icebergView;
+
+  NessieViewOperations(
+      ContentKey key,
+      NessieIcebergClient client,
+      FileIO fileIO,
+      Map<String, String> catalogOptions) {
+    this.key = key;
+    this.client = client;
+    this.fileIO = fileIO;
+    this.catalogOptions = catalogOptions;
+  }
+
+  @Override
+  public void doRefresh() {
+    try {
+      client.refresh();
+    } catch (NessieNotFoundException e) {
+      throw new RuntimeException(
+          String.format(
+              "Failed to refresh as ref '%s' is no longer valid.", client.getRef().getName()),
+          e);
+    }
+    String metadataLocation = null;
+    Reference reference = client.getRef().getReference();
+    try {
+      Content content = client.getApi().getContent().key(key).reference(reference).get().get(key);
+      LOG.debug("Content '{}' at '{}': {}", key, reference, content);
+      if (content == null) {
+        if (currentMetadataLocation() != null) {
+          throw new NoSuchViewException("View does not exist: %s in %s", key, reference);
+        }
+      } else {
+        this.icebergView =
+            content
+                .unwrap(IcebergView.class)
+                .orElseThrow(
+                    () -> {
+                      if (content instanceof IcebergTable) {
+                        return new AlreadyExistsException(
```

Comment:
I think `AlreadyExistsException` makes sense only on commit attempts. When we're simply trying to load a view for a name, but the name maps to a table, returning "nothing" is probably reasonable. I guess, this is what Iceberg core tests expect too.

Reply:
No need to change this as discussed in https://github.com/apache/iceberg/pull/8909#discussion_r1387607715

Reply:
The special case for `IcebergTable` here and for tables doesn't give much benefit. Also that this case throws an `AlreadyExistsException` (valid for commits only) and the `else` below throws and ISE is inconsistent IMO.

Reply:
This particular testcase `createOrReplaceViewThatAlreadyExistsAsTable` from  `ViewCatalogTests` expects this. (similar to other catalogs)

Below ISE, I can change to `AlreadyExistsException`